apps/aecore/src/aec_resilience.erl

```diff
@@ -0,0 +1,121 @@
+-module(aec_resilience).
+-behaviour(gen_server).
+
+%%% This module implements
+
+-export([ start_link/0 ]).
+
+-export([ fork_resistance_active/0
+        ]).
+
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
+
+-record(st, {}).
+
+-define(TAB, ?MODULE).
+
+-type height() :: non_neg_integer().
+-type fork_resistance() :: {yes, height()} | no.
+-type mode() :: manual | auto.
+
+-spec fork_resistance_active() -> fork_resistance().
+fork_resistance_active() ->
+    {Res, _} = get_fork_resistance(),
+    Res.
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init([]) ->
+    ensure_tab(),   % ets table will survive process restart
+    aec_events:subscribe(chain_sync),
+    aec_events:subscribe(update_config),
+    {ok, #st{}}.
+
+handle_call(_Req, _From, St) ->
+    {reply, {error, unknown_call}, St}.
+
+handle_cast(_Msg, St) ->
+    {noreply, St}.
+
+handle_info({gproc_ps_event, chain_sync, #{info := {chain_sync_done, _}}}, St) ->
+    %% For now, at least, we try to enable fork resistance as soon as we are in sync
```

Comment:
Well if the node restarts then it can sync with a malicious chain

Reply:
Yes, but this will need to be prevented using whitelisting

Reply:
Not always - by persisting the time when we were synced it's possible to automatically enforce fork resistance:
If the fork resistance value is 100 and we were offline 30 minutes after restarting(and being synced before) we can still safely set the fork resistance value to 90.

Reply:
Of course when being offline for a long period we need to disable fork resistance. But in most cases crashes/downtimes are small.

Reply:
@uwiger Another possibility:
Instead of keeping the date when we were last "fully synced" we can keep a finalized hash which once enabled by fork resistance must be respected. IMO the whitelist is a countermeasure for when the node is fresh and needs to sync from genesis. After we synced at least once we might be sure that the block at TopHeight-100 is finalized - this should be respected after restarts.